### DISCUSSION TITLE
Codespace creation from branch on template repo defaults to a codespace "created from the template"

### DISCUSSION THREAD

**Discussion by aidandj** [community/community#61200] [Codespaces]
### Select Topic Area

Bug

### Body

When inside a template repository, if you are trying to create a codespace from a branch (New with options... -> Select Branch) it will make a call to `github.com/codespaces` with the form value `codespaces[template_repository_id]` set.

This creates a codespace "From a template" that has no connection back to the repo it was created from. This makes it impossible to create repos on branches through the ui, when developing the template repository.

If you make the same call to `github.com/codespaces` without the `template_repository_id` set, then it works fine.

<img width="731" alt="image" src="https://github.com/community/community/assets/63606283/e0218c1c-fef8-4d54-b614-2cb7b9999f87">

I would expect the above image to create a codespace on a branch in a template repository, but instead i get this:

<img width="406" alt="image" src="https://github.com/community/community/assets/63606283/3d086c6d-f1a5-4597-8e58-248292e46ff8">

The second one in this picture was created by omitting the `template_repository_id` form value, and it successfully opened a codespace on the branch.

<img width="352" alt="image" src="https://github.com/community/community/assets/63606283/b614485b-a566-4f34-85b3-02906cb8b06c">

**Comment by gdomingu** [comment]
Thanks for bringing this up, we will look into it.
